File: rabbit_web/src/utils/ageCalculator.ts
// 计算兔兔当前年龄的工具函数
export const calculateCurrentAge = (rescueDate: string, originalAge: string): string => {
  if (!rescueDate || !originalAge || rescueDate === '未知') {
    return originalAge;
  }

  // 解析救援时间（如"2025年6月"）
  const rescueDateMatch = rescueDate.match(/(\d{4})年(\d{1,2})月?/);
  if (!rescueDateMatch) {
    return originalAge;
  }

  const rescueYear = parseInt(rescueDateMatch[1]);
  const rescueMonth = parseInt(rescueDateMatch[2]);

  // 当前时间：2026年4月
  const currentYear = 2026;
  const currentMonth = 4;

  // 计算经过的月数
  const monthsPassed = (currentYear - rescueYear) * 12 + (currentMonth - rescueMonth);

  // 解析原始年龄
  let totalMonths = 0;

  // 匹配"X岁"
  const yearsMatch = originalAge.match(/(\d+)岁/);
  if (yearsMatch) {
    totalMonths += parseInt(yearsMatch[1]) * 12;
  }

  // 匹配"X个月"
  const monthsMatch = originalAge.match(/(\d+)个月/);
  if (monthsMatch) {
    totalMonths += parseInt(monthsMatch[1]);
  }

  // 加上经过的月数
  totalMonths += monthsPassed;

  // 转换回年龄格式
  const years = Math.floor(totalMonths / 12);
  const months = totalMonths % 12;

  if (years === 0) {
    return `${months}个月`;
  } else if (months === 0) {
    return `${years}岁`;
  } else {
    return `${years}岁${months}个月`;
  }
};

File: Rabbit_iOS/Rabbit_iOS/Base/Helpers/AgeCalculator.swift
//
//  AgeCalculator.swift
//  Rabbit_iOS — 与 rabbit_web/src/utils/ageCalculator.ts 对齐
//

import Foundation

enum AgeCalculator {
    /// 与 Web 一致：当前时间 2026 年 4 月
    private static let currentYear = 2026
    private static let currentMonth = 4

    static func displayAge(registrationDate: String, originalAge: String, status: String) -> String {
        if status == "已去世" {
            return "再也不会老去的天使👼"
        }
        return calculateCurrentAge(rescueDate: registrationDate, originalAge: originalAge)
    }

    static func calculateCurrentAge(rescueDate: String, originalAge: String) -> String {
        if rescueDate.isEmpty || originalAge.isEmpty || rescueDate == "未知" {
            return originalAge
        }
        let pattern = #"(\d{4})年(\d{1,2})月"#
        guard let regex = try? NSRegularExpression(pattern: pattern),
              let match = regex.firstMatch(in: rescueDate, range: NSRange(rescueDate.startIndex..., in: rescueDate)),
              match.numberOfRanges >= 3,
              let yRange = Range(match.range(at: 1), in: rescueDate),
              let mRange = Range(match.range(at: 2), in: rescueDate),
              let rescueYear = Int(rescueDate[yRange]),
              let rescueMonth = Int(rescueDate[mRange])
        else {
            return originalAge
        }

        let monthsPassed = (currentYear - rescueYear) * 12 + (currentMonth - rescueMonth)
        var totalMonths = 0
        let ns = originalAge as NSString
        let full = NSRange(location: 0, length: ns.length)

        if let re = try? NSRegularExpression(pattern: #"(\d+)岁"#),
           let m = re.firstMatch(in: originalAge, range: full),
           m.numberOfRanges > 1 {
            let y = Int(ns.substring(with: m.range(at: 1))) ?? 0
            totalMonths += y * 12
        }
        if let re = try? NSRegularExpression(pattern: #"(\d+)个月"#),
           let m = re.firstMatch(in: originalAge, range: full),
           m.numberOfRanges > 1 {
            totalMonths += Int(ns.substring(with: m.range(at: 1))) ?? 0
        }

        totalMonths += monthsPassed
        let years = totalMonths / 12
        let months = totalMonths % 12

        if years == 0 {
            return "\(months)个月"
        }
        if months == 0 {
            return "\(years)岁"
        }
        return "\(years)岁\(months)个月"
    }
}
